### DISCUSSION TITLE
Ender3 V2/S1 Professional Firmware 20220716

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Ender3V2S1#222] [Announcements]
Universal 20220716 MID-YEAR release
==========================

## What's new in this release ##
* New custom G-code C810 for setting up the toolbar shortcuts
* Enhanced C104
* Editable multiple probing
* Increase step limit to 1999.9
* Screen backlight turns on by rotating the knob encoder
* Automatic recalculation of Mesh insets
* Make PID Plot optional to save flash memory (needs recompile)
* Editable Juntion Deviation (needs recompile with HAS_JUNCTION_DEVIATION)
* Editable Linear advance K parameter (needs recompile with LIN_ADVANCE)
* Host shutdown menu item (needs recompile with HOST_SHUTDOWN_MENU_ITEM enabled)
* Some bugs fixed
* Merged Bugfix 2.1.x branch

 This firmware is based on Marlin, has been optimized and has enhancements for the Creality Ender 3 V2/S1 printer. This is an **universal firmware** that can be used in both RET6 and RCT6 SoC chips that have 256k or 512k Flash memory. 

## Versions
- **Ender 3v2 MM**: if you have the Ender 3v2 near to stock then use this version (Manual Mesh).
- **Ender 3v2 BLT**: use this version if you installed a CR/3D/BLTouch level sensor in the Ender 3v2.
- **Ender 3S1**: use this version if you have an Ender 3S1 printer

## UBL versions
Binaries with UBL in their filename have support for [Marlin Unified Bed Leveling](https://marlinfw.org/docs/features/unified_bed_leveling.html), this system is similar to the bilinear leveling but with a few extra features to provide users with more control. There are multiple guides on the web about UBL, we recommend reading about it before using this new leveling system.

* https://marlinfw.org/docs/features/unified_bed_leveling.html
* https://all3dp.com/2/marlin-ubl-unified-bed-levelin-simply-explained/
* https://www.3dmakerengineering.com/blogs/3d-printing/unified-bed-leveling-marlin
* https://hackaday.com/2022/01/14/3d-printering-getting-started-with-universal-bed-leveling/
  
<img align="left" src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main.jpg" width="270" height="480" />
<img  src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main-iconset.jpg" width="270" height="480" />

The binary files of this release are up to date and you can download them from the assets at the bottom of this page. To know if you have the last release go to the info page in the display of your printer and compare the release build date with the name of the binary file. The filename follows the convention:  
  
Ender3v2-422MM-**YYYYMMDD**.bin
  
Where YYYYMMDD indicates the date of the build. 422 refers to the mainboard version **4.2.2**, 423 refers to the mainboard version **4.2.3**, and 427 is for the **4.2.7** mainboard version, if you got a **4.3.1** board use the 422 version. SKRMINI version is currently for the BTT SKR Mini E3 V3 board (needs a screen adapter).

![Identificando_version_de_placa](https://user-images.githubusercontent.com/2745567/168324748-0c6dca47-41f3-4ad5-9146-a1a3caf6a7ce.jpg)

## Special versions
There are special compiles (as V6/Volcano) here: https://github.com/mriscoc/Marlin_Configurations/releases

## Community links
* [Telegram](https://t.me/ender3v2s1firmware)
* [Reddit](https://www.reddit.com/r/Ender3v2Firmware) 
* [Ender 3V2 Facebook](https://www.facebook.com/groups/ender3v2firmware)
* [Ender 3S1 Facebook](https://www.facebook.com/groups/ender3s1printer)

## Donations
Thank you for your support, I receive donations through [Patreon](https://www.patreon.com/mriscoc) and [Paypal](https://www.paypal.com/paypalme/mriscoc)   

[<img src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif">](https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD)

# Firmware features

* Universal RCT6/RET6 Ender 3 V2/S1 Support
* DWIN / DACAI screens support
* Improved support for Manual Mesh and CR/3D/BLTouch
* User configurable power lost recovery enabler
* Manual and assisted bed tramming
* Tramming Wizard for BL/3D/CRTouch versions
* Long filename support with scroll
* Baby Z-Stepping
* Firmware Retract
* Advanced pause feature
* Configurable park-head position (default x=240 y=220)
* Filament change support
* Filament management (load/unload/park head) in prepare menu
* Print Flow on Motion and Tune menu 
* Home axis after motor disabling
* Fastest Z feed rate
* Live move for all motors but extruder
* Support for S-curve Acceleration
* Reboot printer item on control menu
* Increase extruder feed rate limit to 150 for BMG users
* Allows to invert direction of the extruder on Motion menu
* Support for BTT Smart filament sensor
* Selectable runout filament sensor active state (LOW/HIGH/MOTION)
* Editable runout distance
* Support for µSD extensions
* PID popup with temperature plot
* Lock screen
* Editable physical dimensions
* Mesh viewer
* Custom pre-heat
* Enable/disable beeps sound
* New print g-code preview (use [this ](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview)guide)
* New print done screen
* Two decimal positions for axis coordinates value
* New Z-Offset wizard
* New PID settings menus
* PID moves the head to the center for more accuracy
* Leveling bed temperature linked to bed PID temperature
* Editable extrude minimum temperature
* Brightness settings on tune menu
* Turn off the backlight of the lcd in the tune menu
* New status line system, allows the scroll of large messages
* Home submenu with individual axis homing
* Enabled user setting for 115200 bauds (Creality Box compatibility)
* Head parking before recovery after power failure
* Offset icon changes to show leveling status
* Live end-stop diagnostic page
* Move bed to front after cancelling a print.
* Editable firmware retracts settings for Tune and filament settings menu
* Print Statistics page accessible from the Advanced Settings menu
* Toolbar with 11 configurable shortcuts actions.
* Support in source code of control menus for LED lights as the Neopixel

![latest release](https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/New-beta.gif)

### Manual mesh
* Assisted manual bed tramming
* Assisted manual mesh leveling menu
* Selectable mesh points from 2x2 to 9x9
* Mesh viewer
* Persistent Z-offset
* Move Z to 0 and disable motors option
* Mesh editor for fine tuning

### CR/3D/BLTouch
* Safe Probe Z-Homing
* Bed auto level
* High speed probe
* Retry bed level on error
* Mesh viewer
* Assisted tramming
* Selectable mesh points from 2x2 to 9x9
* Allows the user to cancel an auto leveling process 
* Increase Max Probe Offset and Probing Margin to 60 mm
* M48 Probe Test to Probe Settings menu
* Probe Stow/Deploy in Probe Settings menu
* Option to disable HS mode for better compatibility with CRTouch
* Option to change the Z Probe feed rate
* Configurable Bed auto leveling temperature
* Mesh editor for fine tuning
* Automatic maximize leveling area
* Automatic center leveling area

>Use the 2x2 mesh to level you bed!

<a href="https://youtu.be/qzYJC2AHbW8"><img src="https://img.youtube.com/vi/qzYJC2AHbW8/maxresdefault.jpg" width="640" height="400" /></a>  
  
### G-Codes enhanced support
* Filament Load Unload G-codes (M701/M702)
* Support for M600 filament change
* Support for G-code M117 messages
* Enable M575 change baud rate by G-code
* Extended M75 with filename support, use M75 {{ event.name }} in g-code script
* Enable ARC and 'P' parameter to specify complete circles (G2/G3)
* G35 support
* M155 Geometry report
* Native support of marlin brightness and M256 g-code

### Support for special nonstandard custom C G-Codes
* Implemented C11 En Rn Gn Bn gcode for colorize UI elements
* C108 to cancel screens waiting for user simple confirmation
* C29 F### B### L### R### N# T# Mesh insets, amount of grid points and bed temperature of the mesh leveling
* C100 X### Y### to setup minimum limits
* C101 X### Y### z### to setup maximum limits
* C102 X### Y### to setup bed size
* C104 U1 T### to unlock the hotend maximum temperature to ###°C (CAUTION) 
* C115 Shows ProUI details
* C125 X### Y### z### to setup the park position
* C412 M# set run-out sensor active state (M0:LOW, M1:HIGH, M2:MOTION)
* C510 to lock/unlock the screen (C510 U1 to unlock)
* C562 E# to invert the Extruder (E1 to invert)
* C851 S#### set probe Z feed rate

<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/themes/Select_Colors.jpg" width="270" height="480" /> 

### Advanced settings menu
* Advanced mesh leveling editing features
* Probe offsets menu (BLTouch version)
* Customizable park-head position
* Customizable bed size and z heigh up to 999 x 999 x 999
* Launch PID for nozzle and bed with graphical feedback
* Filament run-out sensor with active state setup (LOW/HIGH)
* Filament load and unload lengths menu items in run-out menu
* Power-loss recovery enable/disable option
* LCD Brightness control
* Select Color Sub Menu
* Beep/Sound enable/disable option
* Selectable mesh point (BLTouch version)
* Lock screen
* End-stop diagnostics

### Special Octoprint / Host print support
* High speed (250000 baud) serial communication
* In screen control of host printing (M73, M75-M77) 
* Display messages on status line (M117)
* Color change of UI elements for visual status report (C11)
* Binary file transfer for remote firmware upgrade
* List binary (.bin) files (M20) for flash firmware support with BFT
* Arc Welder Octoprint plugin support
* EEPROM Editor Octoprint plugin support
* Host action commands for filament runout
* Emergency Parser active
* Advanced OK

**Octoprint bed visualizer support**
<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/Bed-visualizer.png" width="350" height="225" />

### Display enhancements
* Extended DWIN library
* Enhanced status messages
* Improved status bar with temperature, flow, fan, z-offset and xyz positions
* PID auto tune progress on display with plot
* Leveling progress on display
* Boot screen for M997
* Transparent icons for better color theme adaptability
* New print g-code preview
* New print done screen

**A lot of additions and fixes from Bugfix development branch**

## More info on the wiki pages ##
- [**How to install the firmware**](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)

- [**Octoprint settings**](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)

- [**How to make the Manual Mesh**](https://github.com/mriscoc/Ender3V2S1/wiki/Manual-Mesh)

- [**How to install a 3D/BLTouch**](https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch)

- [**How to apply a color theme**](https://github.com/mriscoc/Ender3V2S1/wiki/Color-Themes)

https://user-images.githubusercontent.com/2745567/143180943-f69f5967-34c1-421f-86a1-466ecccab97d.mp4

## Precompiled firmware

<hr /><em>This discussion was created from the release <a href='https://github.com/mriscoc/Ender3V2S1/releases/tag/20220719'>Ender3 V2/S1 Professional Firmware 20220716</a>.</em>

**Comment by Creat-new** [comment]
Thanks again for your efforts, I think your driver aspect is almost perfect, I request to share some different UI configurations (screen colors) to achieve better results @mriscoc

**Reply by mriscoc** [reply]
You can post your own color theme as: https://github.com/mriscoc/Ender3V2S1/issues/208

**Reply by Creat-new** [reply]
thanks

**Reply by Creat-new** [reply]
With this new firmware, I found that there is no way to print silently, where can I fix it? @mriscoc

**Reply by mriscoc** [reply]
Reduce the top speeds and accelerations in Control / Motion.

**Reply by Creat-new** [reply]
thanks

**Reply by Creat-new** [reply]
I also want to ask a question, can I modify the microstep?what s the "standalone mode"? @mriscoc

**Reply by mriscoc** [reply]
That's how the control board communicates and manage the motor drivers.

**Reply by bmtsantos** [reply]
Hello, thanks for this! This has saved my hair when it was released since my BLT experience with stock E3V2 firmware was mostly it failing with a blinking LED, and my prints have been going perfectly ever since.

The one thing I noticed on this new version was a failing filament sensor, which I didn't have, and had to disable in the configuration. Maybe having this disabled by default could be better?

Any chance there could be a build with linear advance enabled? I wanted to try it out, but building this locally opens another can of worms. If this could be added to the prebuilt releases it would nice. 

Thank you for your work!

**Reply by mriscoc** [reply]
Maybe that could be sponsored as special configuration https://github.com/mriscoc/Special_Configurations/releases

**Reply by bmtsantos** [reply]
How would one go about sponsoring this custom build?

**Reply by mriscoc** [reply]
Sponsors are welcome through Patreon and Paypal links in the repository. It is necessary to have a user with the knowledge to make all the hardware mods that the board needs to have linear advance activate in his control board and the values that must be changed in the configuration files, with that data I will open a new release in Special Configurations repository with all the modifications and also compile and post there, the versions that the sponsor needs.

**Comment by mcollins056** [comment]
I recently bought an Ender 3 V2 (V4.2.2) with the CR Touch, and the Creality filament run out sensor. What firmware should I choose?

**Reply by MarkforU** [reply]
[Ender3V2-422-BLTUBL-20220716.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20220716/Ender3V2-422-BLTUBL-20220716.bin)

**Reply by mcollins056** [reply]
My printer is now frozen with the Creality logo upon start-up. How do I factory reset the firmware?

**Reply by mriscoc** [reply]
For factory reset, flash the official Creality firmware and load default settings.

**Comment by Cerberus-EU** [comment]
First of all, thank you very much for this. I makes the printer so much better, but....I dowloaded the latest version (Ender3S1-F1-20220716) and now the printer is not always printing. The jhotend overshoots the temp by 10 degrees C, then cools down and heats up again. Does this over and over without starting the print. After turning the printer on and off, it suddely works, but the next print I have the same issue. 
Sorry I cannot tell you what the trigger is that makes the print start anyway

**Reply by mriscoc** [reply]
Please post the steps to reproduce the issue. Maybe it is a problem with your thermistor.

**Reply by Cerberus-EU** [reply]
Thank you for you're reply. Today I turned the printer on and it worked without a hitch. I will make sure to let you know if it gives an error again. I do not want to go back to the Creality software since yours is much better. I was stunned by the fact that "just" changing the software can open up so much more options. Obviously I am not an engineer, and to old to learn this,  so I am grateful for people like you who are willing to share things like this and put you tome en effort in it.

**Reply by Cerberus-EU** [reply]
Not sure if my earlier post got posted correct.
I turned my printer on today and started a print. Nothing happens. The bed heats up, then the Nozzle. The Nozzle heats up, and cools down till about 10 degrees to low, then heats up again etc etc. No print is started. 
At the bottom of the print screen I see the XYZ, besides the values there I see question marks blinking. So the values are changing with ? 
After several restarts, it is printing now....

**Reply by mriscoc** [reply]
The `--?--` in axes indicators are normal if the motors were disabled or axes were not homing. I think that you can have some connection problem, I don't have that behavior in my printer and I have not received similar report in the test group.

**Reply by Cerberus-EU** [reply]
Well, I will put back the stock Firmware to se what happens. If it works, I will try yours again, because I really like your's much better.
Edit: Just put back the Creality Firmware, works, no ? at the motor axes. I've put on a 12 hour print, so probably tomorrow I will put back your firmware and see what happens

**Comment by MichiMess** [comment]
Thanks again mriscoc! It's not a problem but why are the motors more loud? More steps?

Give this man a beer guys! It's great firmware again!

https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD

**Comment by mriscoc** [comment]
With the latest version ADAPTIVE_STEP_SMOOTHING was disabled for better movement accuracy and to make program memory space for the RCT6 devices, that increase the sound if the steppers move to high speed because that configuration limit the speed.
To return to a more silent operation you can limit the printer speed to the original stock values with:
Control / Motion
Speed menu: X=300, Y=300, Z=5, E=25
Acceleration menu: X=500, Y=500, Z=100, E=1000
Or using the G-codes:

```gcode
M203 X300.00 Y300.00 Z5.00 E25.00 ;Setup machine max feedrate
M201 X500.00 Y500.00 Z100.00 E1000.00 ;Setup machine max acceleration
```

**Reply by MichiMess** [reply]
Ok thanks!!

**Reply by duettra** [reply]
Hi @mriscoc, thanks for this! I am breaking my head around the noise of my E3S1 after the firmware upgrade. Downgraded it, and noise is gone. Re-applied the new firmware even with these M code settings and noise is still present.. 

Here the video: 

01. [Video01 - Issue starts at time of skirt printing](https://drive.google.com/file/d/15liJ-Ze7npMEdo2F069MJ-vnIbNUn-v9/view?usp=sharing)
02. [Video02 - short one](https://drive.google.com/file/d/1wmAjrgJC-t3JgNX2lTyZaB0YUe3dPj98/view?usp=sharing)

Any advise? Using the precompiled firmware binary Ender3S1-F4-20220716.bin.

**Reply by mriscoc** [reply]
Next version will have ADAPTIVE_STEP_SMOOTHING re-enabled to more silent operation but some RCT6 chips will be unsupported due to lack of flash memory space.

**Reply by ghost** [reply]
I compiled by reactivating and I'm using my ender 3 v2

**Comment by meneerneuer** [comment]
Wich of those work on the ender 3 pro with Bltouch, skr mini e3 v3 and sprite extruder. I wanna find a config file to get the right settings for the sprite extruder since the diffrent dimensions are causing a serious headache.

**Reply by mriscoc** [reply]
First, set physical settings, and then probe offset, that's all you need.
https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Comment by gonsan73** [comment]
I have made modifications regarding the hotend, bltouch, fans, etc, in my S1, I have installed the latest Marlin from mriscoc (Miguel Risco-Castillo), but now the printer, when changing the bltouch, does not do me good leveling or the homing. Leave a space to the left of about five centimeters that is not taken into account for leveling. How can I modify it? I know I have to make some changes to Marlin but I don't know exactly how to do it. I'd appreciate some help, please.

**Reply by mriscoc** [reply]
https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Reply by gonsan73** [reply]
Ya, pero es que no sé realmente qué hacer ni qué parámetros cambiar. Además quería instañar un sensor inductivo en vez del bltouch. Puedes ayudarme, por favor? Gracias

**Reply by mriscoc** [reply]
For an EZABL/Autolev sensor you must have a special configuration: https://github.com/mriscoc/Special_Configurations
You can verify if your desired configuration already exists: https://github.com/mriscoc/Special_Configurations/releases

**Comment by zombiejesus123** [comment]
where can i get older versions of your firmware

**Reply by mriscoc** [reply]
https://github.com/mriscoc/Ender3V2S1/releases

**Comment by DoogH2004** [comment]
Hello!  I just bought the Ender 3 V2 Neo.

Ender3V2-422-BLT firmware seems to be working great!

However, I am purchasing a Revo Micro hot end.  This requires me to update some of the config settings.  Can someone provide the config for pre-compiled Ender3V2-422-BLT firmware?

**Reply by DoogH2004** [reply]
@mriscoc

**Reply by mriscoc** [reply]
https://github.com/mriscoc/Special_Configurations

**Comment by kiu345** [comment]
Hello,
has anything changed to the thermal runaway protection in your version or upstream marlin (didn't find anything)? I've migrated the configuration of Ender3V2S1-20220527 (working as expected) to Ender3V2S1-20220716 ("nozzle or bed themperature too high", "Printer killed: MAXTEMP").

Only relevant modification in my configuration is:
TEMP_SENSOR_0 1047

Downgraded to to old one and it is working again, so no sensor error. Any idea why this is happening?

**Reply by mriscoc** [reply]
Currently there is a problem with some sensors that have inverted calibration tables, check here: https://github.com/mriscoc/Ender3V2S1/issues/242

**Comment by RubeNL1992** [comment]
Updated today to version 20220716. But after every few layers it stops printing and got the message Advanced Pause.
Is there a solution for?

**Reply by bmtsantos** [reply]
I had the same issue on a recent previous version, the filament sensor was enabled by default, and I had to disable it in the settings since I don't have one installed.

**Reply by RubeNL1992** [reply]
Readed something similar a while back. But i'm using the filament sensor so that cant be the problem.
Went back to version 202205XX and no problem at all!

**Reply by mriscoc** [reply]
Ender3 V2, S1, 422, 427?

**Reply by RubeNL1992** [reply]
> Ender3 V2, S1, 422, 427?

Ender3 S1

**Reply by mriscoc** [reply]
F1 or F4?

**Reply by RubeNL1992** [reply]
F1

**Reply by sirgrej** [reply]
I have the same issue with the Ender 3 S1. I have checked the runout sensor setting and it is set to "motion". Im not sure whether this is correct.

**Reply by mriscoc** [reply]
Please set runout active to LOW/HIGH

**Reply by mriscoc** [reply]
> I have the same issue with the Ender 3 S1. I have checked the runout sensor setting and it is set to "motion". Im not sure whether this is correct.

MOTION is only correct for BTT Smart Sensor.

**Reply by sirgrej** [reply]
> > I have the same issue with the Ender 3 S1. I have checked the runout sensor setting and it is set to "motion". Im not sure whether this is correct.
> 
> 
> 
> MOTION is only correct for BTT Smart Sensor.

BTT stands for big tree tech? 
Was not familiar with that. 
I will try this setting 

Thanks!

**Reply by sirgrej** [reply]
It worked with setting "high" for the sensor. Don't know whether the runout detaction still works with this setting. I have to test ist separately

**Reply by mriscoc** [reply]
> It worked with setting "high" for the sensor. Don't know whether the runout detaction still works with this setting. I have to test ist separately


Please confirm if the correct settings for Ender3S1 stock filament sensor is HIGH, because Creality in his official repository is LOW:

https://github.com/CrealityOfficial/Ender-3S1/blob/558757cfe542a4e2987b96ef964c98483edc09c9/Marlin/Configuration.h#L1423

**Comment by Creat-new** [comment]
Could you tell me what is the G code to turn off the LCD? @mriscoc

**Reply by mriscoc** [reply]
https://marlinfw.org/docs/gcode/M256.html
